### ISSUE TITLE
System.Runtime.InteropServices.Tests failed in ci

### ISSUE THREAD

**Issue by v-haren** [dotnet/corefx#21703]
/mnt/j/workspace/dotnet_corefx/master/ubuntu16.10_release/Tools/tests.targets(354,5): error : One or more tests failed while running tests from 'System.Runtime.InteropServices.Tests' please check /mnt/j/workspace/dotnet_corefx/master/ubuntu16.10_release/bin/AnyOS.AnyCPU.Release/System.Runtime.InteropServices.Tests/netstandard/testResults.xml for details! [/mnt/j/workspace/dotnet_corefx/master/ubuntu16.10_release/src/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.Tests.csproj]

detail: https://ci.dot.net/job/dotnet_corefx/job/master/job/ubuntu16.10_release/2981/consoleFull#72209969733fe3402-0c0f-45a7-a707-47164965bab5

**Comment by danmosemsft** [comment]
segfault
```
22:33:12   Finished:    System.Runtime.InteropServices.Tests
22:33:15 
  /mnt/j/workspace/dotnet_corefx/master/ubuntu16.10_release/bin/AnyOS.AnyCPU.Release/System.Runtime.InteropServices.Tests/netstandard/RunTests.sh: line 87: 34605 Segmentation fault      (core dumped) $RUNTIME_PATH/dotnet xunit.console.netcore.exe System.Runtime.InteropServices.Tests.dll -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing
```

**Comment by CIPop** [comment]
Failed on Win7 here: https://mc.dot.net/#/user/CIPop/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/536e9bfd4ac7eb248adbd0744cc2bddb8962535b/workItem/System.Runtime.InteropServices.Tests/wilogs

```
2017-06-29 22:31:17,091: INFO: scriptrunner(28): _main: BuildTools Helix Script Runner v0.1 starting
2017-06-29 22:31:17,091: INFO: helix_test_execution(24): __init__: Copying execution payload files from C:\dotnetbuild\work\1a066860-c73b-479a-921f-297efc5fa374\Work\bea9372a-80d9-42bf-86c4-caa4696d6b28\Unzip to C:\dotnetbuild\work\1a066860-c73b-479a-921f-297efc5fa374\Work\bea9372a-80d9-42bf-86c4-caa4696d6b28\Exec\execution
2017-06-29 22:31:17,091: INFO: io(27): copy_tree_to: Copying 'C:\dotnetbuild\work\1a066860-c73b-479a-921f-297efc5fa374\Work\bea9372a-80d9-42bf-86c4-caa4696d6b28\Unzip\RunTests.cmd' to 'C:\dotnetbuild\work\1a066860-c73b-479a-921f-297efc5fa374\Work\bea9372a-80d9-42bf-86c4-caa4696d6b28\Exec\execution\RunTests.cmd'
2017-06-29 22:31:17,091: INFO: io(27): copy_tree_to: Copying 'C:\dotnetbuild\work\1a066860-c73b-479a-921f-297efc5fa374\Work\bea9372a-80d9-42bf-86c4-caa4696d6b28\Unzip\System.Runtime.InteropServices.Tests.dll' to 'C:\dotnetbuild\work\1a066860-c73b-479a-921f-297efc5fa374\Work\bea9372a-80d9-42bf-86c4-caa4696d6b28\Exec\execution\System.Runtime.InteropServices.Tests.dll'
2017-06-29 22:31:17,091: INFO: io(27): copy_tree_to: Copying 'C:\dotnetbuild\work\1a066860-c73b-479a-921f-297efc5fa374\Work\bea9372a-80d9-42bf-86c4-caa4696d6b28\Unzip\System.Runtime.InteropServices.Tests.pdb' to 'C:\dotnetbuild\work\1a066860-c73b-479a-921f-297efc5fa374\Work\bea9372a-80d9-42bf-86c4-caa4696d6b28\Exec\execution\System.Runtime.InteropServices.Tests.pdb'
2017-06-29 22:31:17,107: INFO: io(27): copy_tree_to: Copying 'C:\dotnetbuild\work\1a066860-c73b-479a-921f-297efc5fa374\Work\bea9372a-80d9-42bf-86c4-caa4696d6b28\Unzip\xunit.console.netcore.exe' to 'C:\dotnetbuild\work\1a066860-c73b-479a-921f-297efc5fa374\Work\bea9372a-80d9-42bf-86c4-caa4696d6b28\Exec\execution\xunit.console.netcore.exe'
2017-06-29 22:31:17,107: INFO: io(27): copy_tree_to: Copying 'C:\dotnetbuild\work\1a066860-c73b-479a-921f-297efc5fa374\Work\bea9372a-80d9-42bf-86c4-caa4696d6b28\Unzip\xunit.console.netcore.runtimeconfig.json' to 'C:\dotnetbuild\work\1a066860-c73b-479a-921f-297efc5fa374\Work\bea9372a-80d9-42bf-86c4-caa4696d6b28\Exec\execution\xunit.console.netcore.runtimeconfig.json'
2017-06-29 22:31:17,107: INFO: proc(23): run_and_log_output: Running: C:\dotnetbuild\work\1a066860-c73b-479a-921f-297efc5fa374\Work\bea9372a-80d9-42bf-86c4-caa4696d6b28\Unzip\RunTests.cmd C:\dotnetbuild\work\1a066860-c73b-479a-921f-297efc5fa374\Payload
2017-06-29 22:31:17,107: INFO: proc(26): run_and_log_output: CWD: C:\dotnetbuild\work\1a066860-c73b-479a-921f-297efc5fa374\Work\bea9372a-80d9-42bf-86c4-caa4696d6b28\Unzip
2017-06-29 22:31:17,121: INFO: proc(54): run_and_log_output: Output: Using C:\dotnetbuild\work\1a066860-c73b-479a-921f-297efc5fa374\Payload as the test runtime folder.
2017-06-29 22:31:17,121: INFO: proc(54): run_and_log_output: Output: Executing in C:\dotnetbuild\work\1a066860-c73b-479a-921f-297efc5fa374\Work\bea9372a-80d9-42bf-86c4-caa4696d6b28\Unzip\ 
2017-06-29 22:31:17,121: INFO: proc(54): run_and_log_output: Output: Running tests... Start time: 22:31:17.12
2017-06-29 22:31:17,121: INFO: proc(54): run_and_log_output: Output: 
2017-06-29 22:31:17,121: INFO: proc(54): run_and_log_output: Output: C:\dotnetbuild\work\1a066860-c73b-479a-921f-297efc5fa374\Work\bea9372a-80d9-42bf-86c4-caa4696d6b28\Unzip>call C:\dotnetbuild\work\1a066860-c73b-479a-921f-297efc5fa374\Payload\dotnet.exe xunit.console.netcore.exe System.Runtime.InteropServices.Tests.dll  -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapptests -notrait category=nonwindowstests  -notrait category=OuterLoop -notrait category=failing 
2017-06-29 22:31:17,638: INFO: proc(54): run_and_log_output: Output: xUnit.net console test runner (64-bit .NET Core)
2017-06-29 22:31:17,638: INFO: proc(54): run_and_log_output: Output: Copyright (C) 2014 Outercurve Foundation.
2017-06-29 22:31:17,638: INFO: proc(54): run_and_log_output: Output: 
2017-06-29 22:31:17,888: INFO: proc(54): run_and_log_output: Output: Discovering: System.Runtime.InteropServices.Tests
2017-06-29 22:31:18,357: INFO: proc(54): run_and_log_output: Output: Discovered:  System.Runtime.InteropServices.Tests
2017-06-29 22:31:18,450: INFO: proc(54): run_and_log_output: Output: Starting:    System.Runtime.InteropServices.Tests
2017-06-29 22:31:22,043: INFO: proc(54): run_and_log_output: Output: Finished running tests.  End time=22:31:22.04, Exit code = -1073741819
2017-06-29 22:31:22,043: INFO: proc(58): run_and_log_output: Exit Code: -1073741819
2017-06-29 22:31:22,043: ERROR: scriptrunner(82): _main: Error: No exception thrown, but XUnit results not created
2017-06-29 22:31:22,043: ERROR: helix_test_execution(83): report_error: Error running xunit None

2017-06-29 22:31:22,043: INFO: azure_storage(205): _upload: Uploading single blob: 'error.log'
2017-06-29 22:31:22,184: INFO: saferequests(90): request_with_retry: Response complete with status code '201'
2017-06-29 22:31:22,184: INFO: helix_test_execution(93): report_error: Sending ToF test failure event
2017-06-29 22:31:22,184: INFO: event(38): send: Sending event type XUnitTestFailure
2017-06-29 22:31:22,246: INFO: saferequests(90): request_with_retry: Response complete with status code '201'
2017-06-29 22:31:22,246: INFO: azure_storage(205): _upload: Uploading single blob: '5a7371f2658f4fe9819498394c0ce301.log'
2017-06-29 22:31:22,466: INFO: saferequests(90): request_with_retry: Response complete with status code '201'
2017-06-29 22:31:22,466: INFO: event(38): send: Sending event type Logs
2017-06-29 22:31:22,543: INFO: saferequests(90): request_with_retry: Response complete with status code '201'
```

From the log it appears that the XUnit runner crashed. A memory dump might be available on dumpling.

**Comment by jkotas** [comment]
Likely caused by https://github.com/dotnet/corefx/issues/21741

**Comment by hughbe** [comment]
Fixed in #21743
